File: programming 1/lab 5/src/commands/ExitCommand.java
package commands;

/**
 * Команда для завершения работы приложения.
 */
public class ExitCommand extends Command{
    public ExitCommand () {
        super("exit", "завершить работу");
    }

    /**
     * Выполняет команду без аргументов.
     * @param arg аргумент команды (в данной команде не используется).
     */
    @Override
    public void execute(String arg) {
        if (arg != null) System.err.printf("Команда %s не принимает аргументы\n", getName());
        else {
            System.out.println("Завершение работы...");
            System.exit(0);
        }
    }
}

File: programming 1/lab 7/src/main/java/server/commands/ExitCommand.kt
package server.commands

import common.communication.Response
import common.communication.ResponseStatus
import server.managers.CollectionManager
import server.utils.HistoryWriter

/**
 * Команда для завершения работы приложения.
 */
class ExitCommand(private val cm: CollectionManager) : Command("exit", "завершить работу") {
    /**
     * Выполняет команду без аргументов.
     * @param arg аргумент команды (в данной команде не используется).
     */
    override fun execute(arg: String?): Response {
        require(arg == null) { "Команда $name не принимает аргументы" }
        if (HistoryWriter.file != null) HistoryWriter.clear(HistoryWriter.file!!)
        return Response(ResponseStatus.EXIT, "Завершение работы...")
    }
}
